File: 0706-design-hashmap/0706-design-hashmap.java
class MyHashMap {

    private static final int SIZE = 1000;
    private List<int[]>[] map;

    public MyHashMap() {
        map = new ArrayList[SIZE];
        for (int i = 0; i < SIZE; i++) {
            map[i] = new ArrayList<>();
        }
    }

    public void put(int key, int value) {
        int index = key % SIZE;
        List<int[]> bucket = map[index];
        for (int[] pair : bucket) {
            if (pair[0] == key) {
                pair[1] = value; 
                return;
            }
        }
        bucket.add(new int[]{key, value}); 
    }

    public int get(int key) {
        int index = key % SIZE;
        List<int[]> bucket = map[index];
        for (int[] pair : bucket) {
            if (pair[0] == key) {
                return pair[1]; 
            }
        }
        return -1; 
    }

    public void remove(int key) {
        int index = key % SIZE;
        List<int[]> bucket = map[index];
        for (int i = 0; i < bucket.size(); i++) {
            int[] pair = bucket.get(i);
            if (pair[0] == key) {
                bucket.remove(i); 
                return;
            }
        }
    }
}

File: 0706-design-hashmap/0706-design-hashmap.js
class MyHashMap {
    constructor() {
        this.size = 10;
        this.mp = new Array(this.size).fill(null).map(() => []);
    }

    hash(key) {
        return key % this.size;
    }

    put(key, value) {
        const i = this.hash(key);
        const bucket = this.mp[i];

        for (let j = 0; j < bucket.length; j++) {
            if (bucket[j][0] === key) {
                bucket[j][1] = value;
                return;
            }
        }

        bucket.push([key, value]);
    }

    get(key) {
        const i = this.hash(key);
        const bucket = this.mp[i];

        for (let j = 0; j < bucket.length; j++) {
            if (bucket[j][0] === key) {
                return bucket[j][1];
            }
        }

        return -1;
    }

    remove(key) {
        const i = this.hash(key);
        const bucket = this.mp[i];

        for (let j = 0; j < bucket.length; j++) {
            if (bucket[j][0] === key) {
                bucket.splice(j, 1);
                return;
            }
        }
    }
}

// Usage
const hashMap = new MyHashMap();
hashMap.put(1, 1);
hashMap.put(2, 2);
console.log(hashMap.get(1)); // Output: 1
console.log(hashMap.get(3)); // Output: -1
hashMap.put(2, 1);
console.log(hashMap.get(2)); // Output: 1
hashMap.remove(2);
console.log(hashMap.get(2)); // Output: -1